### PR TITLE
fix: set aiohttp Application client max size according with producer config

### PR DESCRIPTION
**Pull request by jjaakola-aiven**
# About this change - What it does

When setting the producer max request size larger than default the aiohttp Application client max size must be set also larger.

# Why this way

When REST Proxy is enabled the aiohttp client max size selection is adaptive if `aiohttp_client_max_size` configuration is not set. In other cases the default or explicitly set configuration value is used.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -308,6 +308,9 @@ Keys to take special care are the ones needed to configure Kafka and advertised_
    * - Parameter
      - Default Value
      - Description
+   * - ``aiohttp_client_max_size``
```
Comment (aiven-anton) [inline README.rst:311]: question: Shouldn't the usage of aiohttp be kept as an internal implementation detail, that we can change without changing names of configuration parameters?

Comment (jjaakola-aiven) [reply README.rst]: `http_request_max_size`.

File: karapace/config.py [hunk 2]
```diff
@@ -186,6 +187,23 @@ def set_config_defaults(config: ConfigDefaults) -> Config:
...
+        # Always add the aiohttp default client max size as the headroom above the producer max request size.
+        # The input JSON size for REST Proxy is not easy to estimate, lot of small records in single request has
+        # a lot of overhead due to JSON structure.
+        new_config["http_request_max_size"] = new_config["producer_max_request_size"] + DEFAULT_AIOHTTP_CLIENT_MAX_SIZE
+    elif new_config["http_request_max_size"] is None:
+        # Set the default aiohttp client max size
+        new_config["http_request_max_size"] = DEFAULT_AIOHTTP_CLIENT_MAX_SIZE
+    else:
+        # Use the aiohttp client max size from the configuration without altering.
+        pass
```
Comment (aiven-anton) [inline karapace/config.py:205]:
```suggestion
```

suggestion (non-blocking): Skip unnecessary branch.

File: karapace/constants.py [hunk 3]
```diff
@@ -10,6 +10,8 @@
 API_VERSION_AUTO_TIMEOUT_MS: Final = 30000
 TOPIC_CREATION_TIMEOUT_MS: Final = 20000
 DEFAULT_SCHEMA_TOPIC: Final = "_schemas"
+DEFAULT_PRODUCER_MAX_REQUEST: Final = 1048576
+DEFAULT_AIOHTTP_CLIENT_MAX_SIZE = 1024**2
```
Comment (aiven-anton) [inline karapace/constants.py:14]:
```suggestion
DEFAULT_AIOHTTP_CLIENT_MAX_SIZE: Final = 1024**2
```

issue: I assume this one should've also received `Final`.

**Review by aiven-anton — COMMENTED** [review]
Change looks good, left a nitpick and a minor comment.

**Review by tvainika — CHANGES_REQUESTED** [review]
It looks like after iterating with config name you got things broken as `aiohttp.web.Application` is now reading `aiohttp_xxx` config value which was renamed to `http_xxx`.

However I'd like to question whole approach of adding new configuration parameter. What if simply add `aiohttp.web.Application` constructor parameter that is always large enough (based on `producer_max_request_size` setting)? No new config parameter, no need for more complex configuration but simply works?

File: karapace/constants.py (hunk 3 above)
Comment (tvainika) [reply karapace/constants.py:14]: It is a bit confusing that those numerical values are equal to each other, but simply represented differently, one as simple value and other as exponential notation.

File: karapace/config.py (hunk 2 above)
Comment (jjaakola-aiven) [reply karapace/config.py:208]: I can remove, the reason why I added `pass` is to be explicit with the use of the value in the configuration.

**Pull request merged**